icepyx/core/spatial.py

```diff
@@ -269,7 +310,12 @@ def validate_polygon_pairs(spatial_extent):
     return "polygon", polygon, None
 
 
-def validate_polygon_list(spatial_extent):
+def validate_polygon_list(
+    spatial_extent: Union[
+        list[Union[float, str]],
```

Comment:
```suggestion
        list[float],
```
Check me on this please, but I think we're not taking strings for coordinate lists.

Reply:
The `Spatial` class is typed to accept a list of strings representing a bounding box or polygon. The docstring for that states:

> list of coordinates
>             (stored in a list of strings, list of numerics, list of tuples, OR np.ndarray)

Reply:
Below, the docstring for this function also indicates `spatial_extent` takes a list or np.ndarray of strings or numerics.